### PR TITLE
[Bugfix] Resolve MTP > 1 issue when lm head tp > 1

### PR DESCRIPTION
**Pull request by jianzs**
### What this PR does / why we need it?

Previously, the dummy run executed compute_logits only once, regardless of num_speculative_tokens. This caused execute_model to hang on compute_logits when lm head tensor parallelism exceeded 1. The fix ensures compute_logits executes correctly during dummy run, matching num_speculative_tokens.

I set the `non_blocking` argument to False when moving `exceeds_max_model_len` to the CPU. From what I understand, using `non_blocking=True` and immediately accessing the tensor on the CPU can cause accuracy problems. However, this issue doesn't happen when transferring data to a device. ref: https://discuss.pytorch.org/t/should-we-set-non-blocking-to-true/38234/18

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

- vLLM version: v0.11.0
- vLLM main: https://github.com/vllm-project/vllm/commit/2918c1b49c88c29783c86f78d2c4221cb9622379

### CODE REVIEW

**Review by whx-sjtu — APPROVED** [review]
We firstly fixed hanging issue of running MTP=1 with llm head tp in PR #3915. This PR refactors it to run `dummy_compute_logits` in drafter's `dummy_run` and further fixes MTP > 1 scenario. LGTM.

**Comment by zouyida2052** [comment]
I've tested it on deepseek and it proves to be useful, please make ci happy

File: vllm_ascend/spec_decode/mtp_proposer.py [hunk 1]
```diff
@@ -821,7 +824,7 @@ def _propose(
             # For the requests that exceed the max model length, we set the
             # sequence length to 1 to minimize their overheads in attention.
             exceeds_max_model_len_cpu = exceeds_max_model_len.to(
-                attn_metadata_i.seq_lens.device, non_blocking=True)
+                attn_metadata_i.seq_lens.device, non_blocking=False)
```
Comment (zouyida2052) [inline vllm_ascend/spec_decode/mtp_proposer.py]: What does this chang fix? Is it necessary?

Comment (MengqingCao) [inline vllm_ascend/spec_decode/mtp_proposer.py:827]: @jianzs I noticed you explain the reason why we disable non-blocking here. But IMO, the stream will keep the right order of data copy and the following operations in the same stream. I don't get the point on why there is an accuracy issue of this, is this a bug of torch-npu?

Comment (MengqingCao) [reply vllm_ascend/spec_decode/mtp_proposer.py:827]: In discuss offline, @jianzs mentioned that it is fine with h2d non-blocking copy, and the accuracy issue occurs with d2h copy. I think it might be a bug of torch-npu. Thus I'm fine with this change here as a workround, and we'll report this to torch-npu to finally fix it.
